azurerm/internal/services/appplatform/data_source_spring_cloud_config_server.go

```diff
@@ -0,0 +1,174 @@
+package appplatform
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmSpringCloudConfigServer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmSpringCloudConfigServerRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"spring_cloud_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"host_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"host_key_algorithm": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"label": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"repositories": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"host_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"host_key_algorithm": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"label": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password": {
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+						"pattern": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"private_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"search_paths": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"strict_host_key_checking": {
```

Comment:
Could we use `enabled` to be clear here
```suggestion
						"enable_strict_host_key_checking": {
```